### DISCUSSION TITLE
Prometheus exporter based on hurl rules

### DISCUSSION THREAD

**Discussion by Self-Perfection** [Orange-OpenSource/hurl#2361] [Ideas] [answered]
I am considering an idea to describe all the various URLs of my project infrastructure in hurl file to automate testing for changes.

But then it seems to make an extra step and somehow monitor continuously status of the infrastructure. For instance average response time of API endpoints of my project's infra.

Something like prometheus exporter that monitors requests described in hurl file seems a good idea.

Or maybe there is any alternative? Probably a solution that might consume list of targets to monitor that can be created automatically from hurl file?

**Comment by jcamiel** [comment, marked as answer]
Hi @Self-Perfection 

I'm not sure to understand what you're trying to do (I'm not versed with Prometheus).
That said, with [`--json`](https://hurl.dev/docs/manual.html#json), you can run Hurl with and have a JSON output of the run, with [timings](https://hurl.dev/docs/response.html#timings) on each response. You could use these timings metrics to monitor requests for instance.

Another possibility is to use Hurl programmatically (see https://crates.io/crates/hurl). This way you can use Hurl as a Rust library and build upon it a runner to do a more custom run.
